File: alert.py
from datetime import datetime

import utils

class Alert:
    """This class represents an alert and is initialized with
    data in JSON format

    """

    def __init__(self, data):
        self.id             = data['id']
        self.expiry         = datetime.fromtimestamp(data['Expiry']['sec'])
        self.description    = data['MissionInfo']['descText']
        self.location       = data['MissionInfo']['location']
        self.missionType    = data['MissionInfo']['missionType']
        self.faction        = data['MissionInfo']['faction']
        self.reward         = data['MissionInfo']['missionReward']
        self.minLevel       = data['MissionInfo']['minEnemyLevel']
        self.maxLevel       = data['MissionInfo']['maxEnemyLevel']
        self.nightmare      = data['MissionInfo']['nightmare']

    def __str__(self):
        """Returns a string with all the information about this alert

        """
        rewardString = ''

        for item in self.reward['items']:
            rewardString += '{} + '.format(item)

        for item in self.reward['countedItems']:
            rewardString += '{} {} + '.format(item['ItemCount'],
                                              item['ItemType'])

        rewardString += str(self.reward['credits']) + 'cr'

        alertString = ('{0}\n'
                       '{1} ({2})\n'
                       '{3}\n'
                       'level {4} - {5}\n'
                       'Expires in {6}')

        return alertString.format(self.location, self.missionType,
                                  self.faction, rewardString,
                                  self.minLevel, self.maxLevel,
                                  self.get_eta_string())

    def get_eta_string(self):
        """Returns a string containing the alert's ETA

        """
        return utils.timedelta_to_string(self.expiry - datetime.now())

    def get_rewards(self):
        """Returns a list containing the alert's rewards

        """
        return [i for i in self.reward['items']] + \
            [i['ItemType'] for i in self.reward['countedItems']]

File: scripts/lib/alert.js
var util = require('util');
var dsUtil = require('./_utils.js');

var Alert = function(data) {
  this.id = data.id;
  this.activation = new Date(1000 * data.Activation.sec);
  this.expiry = new Date(1000 * data.Expiry.sec);
  this.desctiption = data.MissionInfo.descText;
  this.location = data.MissionInfo.location;
  this.missionType = data.MissionInfo.missionType;
  this.faction = data.MissionInfo.faction;
  this.minLevel = data.MissionInfo.minEnemyLevel;
  this.maxLevel = data.MissionInfo.maxEnemyLevel;
  this.nightmare = data.MissionInfo.nightmare;
  this.archwing = data.MissionInfo.archwingRequired;

  this.items = data.MissionInfo.missionReward.items;
  this.countedItems = data.MissionInfo.missionReward.countedItems;
  this.credits = data.MissionInfo.missionReward.credits;
}

Alert.prototype.toString = function() {
  var rewardString = '';

  for(var i in this.items) {
    rewardString += this.items[i] + ' + ';
  }

  for(var i in this.countedItems) {
    rewardString += util.format('%d %s + ', this.countedItems[i].ItemCount,
				this.countedItems[i].ItemType);
  }

  rewardString += this.credits + 'cr';

  var alertString = util.format('%s\n' +
                                '%s (%s)\n' +
                                '%s\n' +
                                'level %d - %d\n' +
                                'Expires in %s',
                                this.location, this.missionType, this.faction,
                                rewardString, this.minLevel, this.maxLevel,
                                this.getETAString());

  return alertString;
}

Alert.prototype.getETAString = function() {
  return dsUtil.timeDeltaToString(this.expiry.getTime() - Date.now());
}

// Returns an array of String objects with all the rewards of this alert
Alert.prototype.getRewards = function() {
  var rewards = this.items;

  for(var i in this.countedItems) {
    rewards.push(this.countedItems[i].itemType);
  }

  return rewards;
}

Alert.prototype.isExpired = function() {
  return this.expiry.getTime() < Date.now();
}

module.exports = Alert;
